sdk_container/src/third_party/coreos-overlay/sys-apps/ignition/files/0019-Detect-multipart-mime-userdata.patch

```diff
@@ -0,0 +1,291 @@
+From 3624cd1a7f3667a323c3c2379719c272805cf7cd Mon Sep 17 00:00:00 2001
+From: Gabriel Adrian Samfira <gsamfira@cloudbasesolutions.com>
+Date: Wed, 17 May 2023 22:08:24 +0300
+Subject: [PATCH 19/19] Detect multipart mime userdata
+
+Signed-off-by: Gabriel Adrian Samfira <gsamfira@cloudbasesolutions.com>
+---
+ config/util/parsingErrors.go                          | 11 ++++++++++-
+ internal/exec/engine.go                               |  4 ++--
+ .../flatcar/ignition/config/shared/errors/errors.go   |  1 +
+ .../flatcar/ignition/config/v1/cloudinit.go           |  9 +++++++++
+ .../github.com/flatcar/ignition/config/v1/config.go   |  2 ++
+ .../flatcar/ignition/config/v2_0/cloudinit.go         |  9 +++++++++
+ .../github.com/flatcar/ignition/config/v2_0/config.go |  4 +++-
+ .../flatcar/ignition/config/v2_1/cloudinit.go         |  9 +++++++++
+ .../github.com/flatcar/ignition/config/v2_1/config.go |  2 ++
+ .../flatcar/ignition/config/v2_2/cloudinit.go         |  8 ++++++++
+ .../github.com/flatcar/ignition/config/v2_2/config.go |  2 ++
+ .../flatcar/ignition/config/v2_3/cloudinit.go         |  9 +++++++++
+ .../github.com/flatcar/ignition/config/v2_3/config.go |  2 ++
+ .../flatcar/ignition/config/v2_4/cloudinit.go         |  9 +++++++++
+ .../github.com/flatcar/ignition/config/v2_4/config.go |  2 ++
+ 15 files changed, 79 insertions(+), 4 deletions(-)
+
+diff --git a/config/util/parsingErrors.go b/config/util/parsingErrors.go
+index 914bf2bf..ee01481b 100644
+--- a/config/util/parsingErrors.go
++++ b/config/util/parsingErrors.go
+@@ -29,6 +29,15 @@ import (
+ 	"github.com/coreos/vcontext/tree"
+ )
+ 
++func isMultipartMime(userdata []byte) bool {
++	header := strings.SplitN(string(decompressIfGzipped(userdata)), "\n", 2)[0]
++
++	// Trim trailing whitespaces
++	header = strings.TrimRightFunc(header, unicode.IsSpace)
++
++	return strings.Contains(header, "Content-Type: multipart/mixed;")
++}
++
+ func isCloudConfig(userdata []byte) bool {
+ 	header := strings.SplitN(string(decompressIfGzipped(userdata)), "\n", 2)[0]
+ 
+@@ -62,7 +71,7 @@ func decompressIfGzipped(data []byte) []byte {
+ func HandleParseErrors(rawConfig []byte, to interface{}) (report.Report, error) {
+ 	r := report.Report{}
+ 
+-	if isCloudConfig(rawConfig) || isScript(rawConfig) {
++	if isCloudConfig(rawConfig) || isScript(rawConfig) || isMultipartMime(rawConfig) {
```

Comment:
Or maybe is it enough to do it only here and the changes in bendor are not needed?

Reply:
This is indeed the only needed change.

Reply:
updated. PTAL.